src/puppet-dom.js

```diff
@@ -14,7 +14,10 @@
     var onDataReady = options.callback;
     this.element = options.listenTo || document.body;
     var clickHandler = this.clickHandler.bind(this);
-    var historyHandler = this.historyHandler.bind(this);
+    this.historyHandler = this.historyHandler.bind(this);
+
+    /* no need to wait for data to be ready to handle callback */
+    this.element.addEventListener('puppet-redirect-pushstate', this.historyHandler);
```

Comment:
After this change, calling `this.unlisten(); this.listen();` will make all the handlers listening except `puppet-redirect-pushstate`.

Perhaps a more adequate solution would be to call `this.listen()` at the end of the constructor.

But still I think that it is not the perfect solution because `this.network.changeState(location.href);` will fail if called before the initial response `obj` is loaded from the server.

I just wonder why do you think that the fix should be applied in PuppetDOM constructor, not in `bootstrap` as I proposed (https://github.com/Palindrom/palindrom-redirect/issues/4#issuecomment-282276197)? Does that not work for you?